src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java

```diff
@@ -506,22 +604,19 @@ public String getNucleusVersion() {
      *
      * @return version from the zip file, or a default if the version can't be determined
      */
-    public static String getVersionFromBuildMetadataFile() {
+    public static String getVersionFromBuildRecipeFile() {
         try {
-            try (InputStream is = Files
-                    .newInputStream(locateCurrentKernelUnpackDir().resolve(NUCLEUS_BUILD_METADATA_DIRECTORY)
-                            .resolve(NUCLEUS_BUILD_METADATA_FILENAME))) {
-                NUCLEUS_BUILD_PROPERTIES.load(is);
-            }
-
-            String version = NUCLEUS_BUILD_PROPERTIES.getProperty(NUCLEUS_VERSION_BUILD_METADATA_KEY);
-            if (version != null) {
-                return version;
+            com.amazon.aws.iot.greengrass.component.common.ComponentRecipe recipe = getRecipeSerializer()
+                    .readValue(locateCurrentKernelUnpackDir().resolve(NUCLEUS_BUILD_METADATA_DIRECTORY)
+                                    .resolve(NUCLEUS_RECIPE_FILENAME).toFile(),
```

Comment:
We need to modify the pipeline to write the version number into the recipe now instead of properties?

Reply:
Yes. Two options
1. Update version in recipe
2. Keep build.properties and here we interpolate the recipe with correct version

I'm currently going with #1 but it can be changed